File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> indexMap = new HashMap<>();
        int[] dummy = {0,0};
        for (int i = 0; i < nums.length; i++){
            int difference = target - nums[i];
            if (indexMap.containsKey(difference)){
                int[] indices = {indexMap.get(difference), i};
                return indices;
            } else{
                indexMap.put(nums[i], i);
            }
        }
        return dummy;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        hash_table = {}
        for num in nums:
            hash_table[num] = 0
        for num in nums:
            if hash_table[num] == 1:
                return True
            else:
                hash_table[num] = 1
        return False
        